### PR TITLE
Containerised sia setup

### PR DESCRIPTION
**Pull request by kwypchlo**
_work in progress_

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -15,6 +15,23 @@ services:
     networks:
       - shared
 
+  sia:
+    image: nebulouslabs/sia:dev
```
Comment (ro-tex) [inline docker-compose.yml:19]: This needs to be the stable image. I guess we're gonna switch to a tag as soon as we have one.

File: docker/sia/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+FROM golang AS builder
+LABEL maintainer="NebulousLabs <devs@nebulous.tech>"
+
+ENV GOOS linux
+ENV GOARCH amd64
+
+RUN git clone https://gitlab.com/NebulousLabs/Sia.git && \
+    cd Sia && \
+    git checkout siad-v150-pre-rc && \
+    make release
```
Comment (ro-tex) [inline docker/sia/Dockerfile:10]: This whole file should be gone - we should tag the RC, publish an image and use that one.

**Comment by m-cat** [comment]
Reminder to close https://gitlab.com/NebulousLabs/Sia/-/issues/3768 once this is merged.

File: docker-compose.yml (hunk 1 above)
Comment (kwypchlo) [reply docker-compose.yml]: using a builder here with v1.5.0 branch

**Pull request merged**